contracts/modules/v2/StakingPool.sol

```diff
@@ -51,11 +52,12 @@ contract StakingPool is IStakingPool, ERC721 {
   // applies to active stake only and does not need update on deposits
   uint public rewardPerSecond;
 
-  // accumulated reward per share of pool
-  uint public accRewardPerPoolShare;
+  // TODO: this should be allowed to overflow (similar to uniswapv2 twap)
+  // accumulated rewarded nxm per reward share
+  uint public accNxmPerRewardsShare;
 
-  // last accRewardPerPoolShare update timestamp
-  uint public lastRewardUpdate;
+  // timestamp when accNxmPerRewardsShare was last updated
+  uint public lastAccNxmUpdate;
 
   uint public firstActiveGroupId;
```

Comment:
not clear to me if firstActiveGroupId is initialized anywhere when the pool starts working

Reply:
This change won't bring significant benefits and does not change the overall mecanics in any way. Given the derivation of the group and bucket ids are done in a lot of places this change would imply a change that increases the chance of human errors and it doesn't seem to be worth implementing.